bootloader/src/main.c

```diff
@@ -414,8 +417,19 @@ int main(void)
             // Initialize user application's Stack Pointer
             __set_MSP(*(__IO uint32_t*) ApplicationAddress);
 
-            // Set IWDG Timeout to 5 secs based on platform specific system flags
-            IWDG_Reset_Enable(5 * TIMING_IWDG_RELOAD);
+            uint8_t disable_iwdg = 0;
+#ifdef CHECK_FIRMWARE
+            // Pre-0.7.0 firmwares were expecting IWDG flag to be set in the DCT, now it's stored in
+            // the backup registers. As a workaround, we disable IWDG if an older firmware is detected
+            const uint16_t module_ver = get_main_module_version();
+            if (module_ver >= 0 && module_ver < SYSTEM_MODULE_VERSION_0_7_0_RC1) {
```

Comment:
> Should this be MIN_MODULE_VERSION_SYSTEM_PART2 instead of SYSTEM_MODULE_VERSION_0_7_0_RC1?

No, `MIN_MODULE_VERSION_SYSTEM_PART2` is internal for `bootloader_dct.c`.

> Also should module_ver be declared as const int16_t to match return sign from get_main_module_version()?

Yeah, `module_ver` should be a signed integer, thanks. I first used 0 as error code, but then it appeared that 0 is actually used as a version number for all current monolithic builds.

> Part of this question is correct logic, and other part what should the default be in case of module corruption? Seems like we'd need to disable the IWDG to be able to DFU new firmware, so we'd want to always default to IWDG disabled.

This code actually doesn't check integrity of any system modules, it only checks version of the main system module (which is either part2 or monolithic firmware module) to workaround particular issue with downgrading from 0.7.0. There's an existing logic in the bootloader, which turns the device into DFU mode once a specific number of IWDG resets is reached, though I never seen it in action.